Mwd

maciek001 opened the pull request:
Dorzucam dwa pliki MWD.h i MWD.cpp i kilka linijek kodu do Console, Globals i USEFILE do EU07
Proponuje to zbudować na jakiejś kopii i sprawdzić. U mnie się kompiluje i działa ;)

firleju commented on Console/MWD.cpp:
```diff
@@ -0,0 +1,678 @@
+
+bool MWDComm::Close()		// zamykanie portu COM
+{
+    WriteLog("zamykam COM-a");
```
Logi po angielsku

firleju commented on Console/MWD.cpp:
```diff
@@ -0,0 +1,678 @@
+    SendData();
+    Sleep(700);
+    CloseHandle(hComm);
+    WriteLog("COM zamkniety");
```
Logi po angielsku

firleju commented on Console/MWD.cpp:
```diff
@@ -0,0 +1,678 @@
+    Sleep(100);
+    SendData();
+    Sleep(700);
+    CloseHandle(hComm);
```
Nie do końca wiem jak działa COM, ale czy nie ma tam specjalnych ramek odpowiedzialnych za zarządzanie połączeniem? W ogóle czy jest sens wysyłanie czegokolwiek jeśli potem wyłączasz fizycznie pulpit i tak zostanie wszystko wyzerowane samo przez się.

firleju commented on Console/MWD.cpp:
```diff
@@ -0,0 +1,678 @@
+{
+    WriteLog("zamykam COM-a");
+    int i = 0;
+    while(i<BYTETOWRITE)
```
Jeśli zamykasz połączenie jest sens zerowania tego co jest w buforze? Przecież i tak tego nie użyjesz bo niszczysz obiekt i bufor zostanie zwolniony z pamięci.

firleju commented on Console/MWD.cpp:
```diff
@@ -0,0 +1,678 @@
+    return TRUE;
+}
+
+bool MWDComm::GetMWDState()	// sprawdzanie otwarcia portu COM
```
To powinno być inline

firleju commented on Console/MWD.cpp:
W jakim celu funkcja zwraca bool jeśli zawsze jest to true?
Może zamiast int s dać bezpośrednie użycie keyeventa?

firleju commented on Console/MWD.cpp:
Tutaj w każdym użyciu funkcji jest niepotrzebne rzutowanie typu int na bool.

firleju commented on Console/MWD.cpp:
Jeśli zamiast int s dać enum z dwFlags będzie można ustawiać więcej rożnych zmiennych i nie robić brancha przy okazji. Ewentualnie zamiast if dać switch-a.

firleju commented on Console/MWD.cpp:
```diff
@@ -0,0 +1,678 @@
+
+    // wciskanie przycisk�w klawiatury
+    /*PORT0*/
+    if(maskData[0] & 0x02) if(lastStateData[0] & 0x02) 	KeyBoard('M',1);	// wy��czenie wy��cznika szybkiego
```
Czy to ma działać tak, że włączamy po drugim przeczytaniu informacji a wyłączamy po pierwszym przeczytaniu? To nie reaguje na zmiany stanu. Tyczy się wszystkich warunków niżej.
I nie widzę sensu w robieniu dwóch if-ów jeśli można zrobić jednym.

firleju commented on Console/MWD.h:
```diff
@@ -0,0 +1,93 @@
+#ifndef MWDH
```
Na początku powinna być informacja o licencji.

firleju commented on Console/MWD.cpp:
```diff
@@ -0,0 +1,678 @@
+/*
```
Na początku powinna być informacja o licencji.

firleju reviewed (changes requested):
Moje komentarze do kodu w treści.

maciek001 committed:
1. Licencje dodane - jeżeli uważasz, że źle to daj znać jak to naprawić ;)
2. Przy zamykaniu połączenia ustawiam zera i wysyłam żeby cały pulpit się wyłączył (lampki, kontrolki, buczki), hasler przestał kręcić itd. 
3. Ramek do sterowania nie ma: sama specyfikacja RS-232 określa tylko ramkę danych (nie ma rozróżneinia między danymi a sterowaniem) i sygnałami sterującymi takimi jak RTS, CTS. Na razie nie wykorzystuję RTS ale prawdopodobnie będę go dorzucał jako dodatkowy sygnał.
4. Emulacja klawiatury: najpierw "wciskam" przycisk na klawiaturze i żeby nie generować opuźnienia puszczam go dopiero w następnej obsłudze. Obsługa wszystkiego na raz (wciskanie i puszczanie) nie działa bo za szybko się to robi ;)
5. Funkcja emulacji klawiatury działa następująco: przesyłasz numer przycisku do wciśnięcia i (teraz już boola) czy ma być wciśnięty czy puszczony.

maciek001 committed:
zmiana podwójnych if-ów na &&

maciek001 committed:
poprawka przeliczania wartości analogowej

maciek001 committed:
Dodana blokada odbioru danych (wpis w pliku eu07.ini) UWAGA! nie testowane bo nie mogę otworzyć projektu.
